packages/react-components/react-field/src/contexts/useFieldContextValues.ts

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+
+import type { FieldContextValue, FieldContextValues, FieldState } from '../Field';
+
+/**
+ * Get the context values used when rendering Field.
+ */
+export const useFieldContextValues_unstable = (state: FieldState): FieldContextValues => {
+  const { generatedControlId, orientation, required, size, validationState } = state;
+  const labelFor = state.label?.htmlFor;
+  const labelId = state.label?.id;
+  const validationMessageId = state.validationMessage?.id;
+  const hintId = state.hint?.id;
+
+  const field: FieldContextValue = React.useMemo(
+    () => ({
+      generatedControlId,
+      hintId,
+      labelFor,
+      labelId,
+      orientation,
+      required,
+      size,
+      validationMessageId,
+      validationState,
+    }),
+    [generatedControlId, hintId, labelFor, labelId, orientation, required, size, validationMessageId, validationState],
+  );
+
+  return { field };
```

Comment:
Should `{ field }` be what is returned from the `useMemo()` call above? Right now this hook is returning a new object on every call which seems to defeat the purpose of memoizing?

Reply:
This is how contexts work in our controls. It's a bit confusing. This function is returning an object that contains all of the context values used by Field (which is just one: `field`). When rendering, the memoized `field` value is what's passed to the context.